databricks_cli/libraries/cli.py

```diff
@@ -0,0 +1,263 @@
+              help=PYPI_PACKAGE_HELP)
+@click.option('--pypi-repo', help=PYPI_REPO_HELP)
+@click.option('--cran-package', cls=MutuallyExclusiveOption, mutually_exclusive=INSTALL_OPTIONS,
+              help=CRAN_PACKAGE_HELP)
+@click.option('--cran-repo', help=CRAN_REPO_HELP)
+@require_config
+@eat_exceptions # noqa
+def install_cli(cluster_id, jar, egg, maven_coordinates, maven_repo, maven_exclusion, pypi_package, # noqa
+                pypi_repo, cran_package, cran_repo):
+    """
+    Install libraries on a cluster. Libraries must be first uploaded to dbfs or s3
+    (see `dbfs cp -h`). Unlike the API, only one library can be installed for each execution of
+    `databricks libraries install`.
+
+    Only provide one of [--jar, --egg, --maven-coordinates, --pypi-package, --cran-package].
+    """
+    maven_exclusion = list(maven_exclusion)
+    libraries = []
+    if jar is not None:
+        libraries.append({'jar': jar})
+    elif egg is not None:
+        libraries.append({'egg': egg})
+    elif maven_coordinates is not None:
+        maven_library = {'maven': {'coordinates': maven_coordinates}}
+        if maven_repo is not None:
+            maven_library['maven']['repo'] = maven_repo
+        if len(maven_exclusion) > 0:
+            maven_library['maven']['exclusions'] = maven_exclusion
+        libraries.append(maven_library)
+    elif pypi_package is not None:
+        pypi_library = {'pypi': {'package': pypi_package}}
+        if pypi_repo is not None:
+            pypi_library['pypi']['repo'] = pypi_repo
+        libraries.append(pypi_library)
+    elif cran_package is not None:
+        cran_library = {'cran': {'package': cran_package}}
+        if cran_repo is not None:
+            cran_library['cran']['repo'] = cran_repo
+        libraries.append(cran_library)
+    assert len(libraries) == 1, 'Should have one library to install.'
+    install_libraries(cluster_id, libraries)
+
+
+def _uninstall_cli_exit_help(cluster_id):
+    click.echo("WARNING: Uninstalling libraries requires a cluster restart.")
+    click.echo("databricks clusters restart --cluster-id {}".format(cluster_id))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Uninstall libraries on a cluster.')
+@click.option('--cluster-id', required=True, type=ClusterIdClickType(),
+              help=ClusterIdClickType.help)
+@click.option('--all', is_flag=True, default=False, help='If set, uninstall all libraries.')
+@click.option('--jar', cls=MutuallyExclusiveOption, mutually_exclusive=INSTALL_OPTIONS,
+              help=JAR_HELP)
+@click.option('--egg', cls=MutuallyExclusiveOption, mutually_exclusive=INSTALL_OPTIONS,
+              help=EGG_HELP)
+@click.option('--maven-coordinates', cls=MutuallyExclusiveOption,
+              mutually_exclusive=INSTALL_OPTIONS, help=MAVEN_COORDINATES_HELP)
+@click.option('--maven-repo', help=MAVEN_REPO_HELP)
```

Comment:
Is it needed? What happens if I installed one library with non-default repo and then tried uninstalling it only using the coordinates?

Reply:
I just checked. Unfortunately it is needed.